### PR TITLE
feat(artwork): Add Video support to carousel

### PR DESCRIPTION
**Pull request by damassi**
This PR resolves [MOPLAT-544]

### Description

This adds new Video support to the Artwork page, and more specifically the `ImageCarousel` component. Along the way a couple things had to happen:
- `ImageCarousel` needed support for video, which was added via [`react-native-vimeo-iframe`](https://github.com/MetaLabs-inc/react-native-vimeo-iframe)
- `ImageCarousel` needed to be updated to use the more up-to-date `figures` field from MP; this allows us to discriminate on the `Video` type for artwork media 
- Merge `Image` and `Video` types from MP into a new `media` prop that the `ImageCarousel` accepts; and lastly
- Rename the `images` prop to something more specific to how its used outside of a MP context: `staticImages`. 

**To test:**
iOS:
- `yarn navigate-to-route /artwork/rodrigo-rodriguez-tranformacion`
- `yarn navigate-to-route /artwork/sadaharu-horio-ku-wei-zhen-zhi-untitled-73`
- `yarn navigate-to-route /artwork/eva-evangelista-window`  

Android:
- `yarn navigate-to-route-android /artwork/rodrigo-rodriguez-tranformacion`
- `yarn navigate-to-route-android /artwork/sadaharu-horio-ku-wei-zhen-zhi-untitled-73`
- `yarn navigate-to-route-android /artwork/eva-evangelista-window`  

**Additional QA:**
- [x] Get Android sim working and test there
- [ ] Check out MyCollection header carousel (can't figure out how to navigate to in-app)
- [ ] Check out ViewingRoom works carousel (also can't figure out) 

https://user-images.githubusercontent.com/236943/215013357-305a4f49-1c1a-460d-ad5e-7751187fc2a5.mp4

**Android:**

https://user-images.githubusercontent.com/236943/215243265-ee208a96-aa20-4237-8744-bc71cddaa4a2.mp4

cc @artsy/mobile-platform 

### PR Checklist

- [x] I tested my changes on **iOS**.
- [x] I tested my changes on **Android**.
- [x] I added screenshots or videos to illustrate my changes.
- [x] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].
- [ ] I have prefixed changes that need to be tested during a release QA with **[NEEDS EXTERNAL QA]** on the changelog.

### To the reviewers 👀

- [x] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Added support for viewing artwork page videos 

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[MOPLAT-544]: https://artsyproduct.atlassian.net/browse/MOPLAT-544?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/Artwork/Components/ArtworkHeader.tsx [hunk 1]
```diff
@@ -54,7 +55,8 @@ export const ArtworkHeader: React.FC<ArtworkHeaderProps> = (props) => {
   const showWhatsAppItem = useCanOpenURL("whatsapp://send?phone=+491898")
   const showInstagramStoriesItem = useCanOpenURL("instagram://user?username=instagram")
 
-  const currentImage = (artwork.images ?? [])[currentImageIndex]
+  const imageFigures = artwork.images.filter(guardFactory("__typename", "Image"))
```
Comment (damassi) [inline src/app/Scenes/Artwork/Components/ArtworkHeader.tsx:58]: fixme

File: src/app/Scenes/Artwork/Components/ImageCarousel/ImageCarouselContext.tsx [hunk 2]
```diff
@@ -70,6 +83,9 @@ export function useNewImageCarouselContext({
   const [isZoomedCompletelyOut, setIsZoomedCompletelyOut] = useGlobalState(true)
   const tracking = useTracking()
 
+  const media = setVideoAsCover ? [...videos, ...images] : [...images, ...videos] ?? []
+  // const media = [...videos, ...images]
```
Comment (damassi) [inline src/app/Scenes/Artwork/Components/ImageCarousel/ImageCarouselContext.tsx:87]: fixme

File: src/app/Scenes/ViewingRoom/ViewingRoomArtwork.tsx [hunk 3]
```diff
@@ -266,8 +269,8 @@ const selectedArtworkFragmentSpec = graphql`
     widthCm
     heightCm
     id
-    images {
-      ...ImageCarousel_images @relay(mask: false) # We need this because ImageCarousel uses regular react-relay and we have relay-hooks here.
+    figures {
+      ...ImageCarousel_figures @relay(mask: false) # We need this because ImageCarousel uses regular react-relay and we have relay-hooks here.
```
Comment (damassi) [inline src/app/Scenes/ViewingRoom/ViewingRoomArtwork.tsx:273]: todo: confirm this comment

File: src/app/Scenes/Artwork/Components/ImageCarousel/ImageCarousel.tsx [hunk 4]
```diff
@@ -45,73 +50,30 @@ export interface ImageCarouselProps {
...
-        }
-      }
-    }
-
-    return result
-  }, [props.images])
-
-  const context = useNewImageCarouselContext({ images, onImageIndexChange })
+  const { cardHeight, onImageIndexChange, setVideoAsCover } = props
+  const { images, videos, disableDeepZoom } = useImageCarouselMedia(props)
```
Comment (damassi) [inline src/app/Scenes/Artwork/Components/ImageCarousel/ImageCarousel.tsx:54]: Moved all of this logic into a hook to try to clean up the body of the component

File: src/app/Scenes/Artwork/Components/ImageCarousel/ImageCarousel.tsx [hunk 5]
```diff
@@ -133,24 +95,36 @@ export const ImagesCarousel = () => {
...
-        image: Image {
-          tileSize: TileSize
-          url: Url
-          format: Format
-          size: Size {
-            width: Width
-            height: Height
+  figures: graphql`
+    fragment ImageCarousel_figures on ArtworkFigures @relay(plural: true) {
+      ... on Image {
```
Comment (damassi) [inline src/app/Scenes/Artwork/Components/ImageCarousel/ImageCarousel.tsx:100]: Note the union type

File: src/app/Scenes/Artwork/Components/ImageCarousel/ImageCarousel.tsx [hunk 6]
```diff
@@ -133,24 +95,36 @@ export const ImagesCarousel = () => {
...
+            url: Url
+            format: Format
+            size: Size {
+              width: Width
+              height: Height
+            }
           }
         }
       }
+      ... on Video {
```
Comment (damassi) [inline src/app/Scenes/Artwork/Components/ImageCarousel/ImageCarousel.tsx:120]: And here

File: src/app/Scenes/Artwork/Components/ImageCarousel/ImageCarousel.tsx [hunk 7]
```diff
@@ -188,6 +162,106 @@ function getBestImageVersionForThumbnail(imageVersions: readonly string[]) {
...
+  const embeddedCardBoundingBox = {
+    width: screenDimensions.width,
+    height: isPad() ? 460 : props.cardHeight,
+  }
+
+  const imageFigures = props.staticImages?.length
+    ? props.staticImages
+    : props.figures?.filter(guardFactory("__typename", "Image"))
+
+  const videoFigures = props.figures?.filter(guardFactory("__typename", "Video"))
```
Comment (damassi) [inline src/app/Scenes/Artwork/Components/ImageCarousel/ImageCarousel.tsx:187]: Handy util for filtering on unions, which is typically a ts headache

File: src/app/Scenes/Artwork/Components/ImageCarousel/ImageCarousel.tests.tsx [hunk 8]
```diff
@@ -12,126 +12,24 @@ import {
 } from "./ImageCarousel"
 import { getMeasurements } from "./geometry"
 
-
-const deepZoomFixture: NonNullable<
```
Comment (damassi) [inline src/app/Scenes/Artwork/Components/ImageCarousel/ImageCarousel.tests.tsx:16]: moved fixtures to the bottom

File: src/app/Scenes/Artwork/Components/ImageCarousel/ImageCarousel.tests.tsx [hunk 9]
```diff
@@ -313,60 +201,239 @@ describe("ImageCarouselFragmentContainer", () => {
...
       })
 
       expect(queryAllByLabelText("Image Pagination Indicator")).toHaveLength(0)
     })
 
     it("disables scrolling", async () => {
-      const { getByLabelText } = renderWithWrappers(<TestWrapper />)
-
-      resolveMostRecentRelayOperation(mockEnvironment, {
+      const { getByLabelText } = renderWithRelay({
```
Comment (damassi) [inline src/app/Scenes/Artwork/Components/ImageCarousel/ImageCarousel.tests.tsx:216]: Updating old relay code to use our test helpers is so easy :)

File: typings/react-native-config.d.ts [hunk 10]
```diff
@@ -16,5 +16,6 @@ declare module "react-native-config" {
     UNLEASH_PROXY_CLIENT_KEY_STAGING: string
     UNLEASH_PROXY_URL_PRODUCTION: string
     UNLEASH_PROXY_URL_STAGING: string
+    VIMEO_PUBLIC_TOKEN: string
```
Comment (pvinis) [inline typings/react-native-config.d.ts:19]: i think this should be added to `.env.example` too. and then also make sure to run the `yarn setup:artsy:update!` script too to upload the token to s3

**Review by pvinis — DISMISSED** [review]
nice. one comment about the env var, otherwise 👍. the image carousel is not the friendliest part of eigen, so well done.

assigning back to you.

**Pull request merged**